py3status/modules/sysdata.py

```diff
@@ -309,16 +301,8 @@ def sysdata(self):
         sys = {'temp_unit': self.temp_unit}
         # get CPU usage info
         if self.py3.format_contains(self.format, 'cpu_usage'):
-            cpu_total, cpu_idle = self._get_cpu()
-            cpu_usage = 0
-            if cpu_total != self.cpu_total:
-                cpu_usage = (1 - (
-                    float(cpu_idle - self.cpu_idle) / float(cpu_total - self.cpu_total)
-                )) * 100
-            sys['cpu_usage'] = cpu_usage
-            self.cpu_total = cpu_total
-            self.cpu_idle = cpu_idle
-            self.py3.threshold_get_color(cpu_usage, 'cpu')
+            sys['cpu_usage'] = self._calc_cpuinfo(self._get_stat())
```

Comment:
>Why are we passing self._get_stat()

An attempt to make a consistency out of everything in this module. For Mem + Swap, we get `memi` then pass it in for `mem` and `swap`. With `cpu`, we can change to what you said above.

If somebody made a new `cpu` function, it may be in either `_calc_cpuinfo` or in a new function. Tbh, `calc_cpuinfo` might be a poor name... I think maybe change it to `_calc_cpu_percent` or `_get_cpu_percent` would be better... then one day when we have more `cpu`-related functions, `_get_stat()` would be moved out and be replaced with assigned `stat`.

I plan to rename `cpu_usage` to `cpu_percent` too. Lmk what you want to do here.

Reply:
Looking around, I can make a new cpu placeholder with new cpu function using  `/proc/cpuinfo`, but if it was `/proc/stat`, we would want to read `/proc/stat` once and pass `stat` in to those who need it... similar to `mem` and `swap`. Hope you merge this soon. Xo.